napari/layers/vectors/vectors.py

```diff
@@ -436,6 +443,22 @@ def edge_width(self, edge_width: Union[int, float]):
         self.events.edge_width()
         self.refresh()
 
+    @property
+    def edge_display_style(self) -> str:
+        """str: Vectors display style."""
+        return self._edge_display_style
+
+    @edge_display_style.setter
+    def edge_display_style(self, edge_display_style: str):
+        if edge_display_style in ['rectangle', 'triangle', 'arrow']:
+            # There should probably be a 'DisplayStyle' class
+            self._edge_display_style = edge_display_style
+        else:
+            raise
+
+        self.events.edge_display_style()
+        self.refresh()
```

Comment:
these should only be called if the style actually changes from before... though I see now that many other places don't do that :/

Reply:
Do you mean that I should add a test like
```
old_style = self._edge_display_style

if old_style != edge_display_style:
    if edge_display_style in ['rectangle', 'triangle', 'arrow']:
         self._edge_display_style = edge_display_style
    else:
         raise
    self.events.edge_display_style()
    self.refresh()
```
I don't see any such test, for example for the blending mode. Have I misunderstood ?

Reply:
Yeah, indeed this is missing for other places, but it shouldn't :P

I would do something like this (assuming a `VectorStyle` enum like I suggested above):

```py
    @edge_display_style.setter
    def edge_display_style(self, edge_display_style: str):
        old_style = self._edge_display_style
        self._edge_display_style = VectorStyle(edge_display_style) # this already takes care of checking and raising exceptions
        if self._edge_display_style != old_style:
            self.events.edge_display_style()
            self.refresh()
```